core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala

```diff
@@ -496,7 +517,7 @@ class FsHistoryProviderSuite extends SparkFunSuite with BeforeAndAfter with Matc
 
       var provider: FsHistoryProvider = null
       try {
-        provider = new FsHistoryProvider(conf)
+        provider = newProvider(conf)
```

Comment:
it seems worse to have `newProvider` method and create a lof of diff.

Reply:
This was part of some code I only partially reverted. Will revert these.